org-formation/organization.yaml

```diff
@@ -314,6 +315,7 @@ Organization:
         <<: !Include ./_default_org_tags.yaml
         Department: CNB
         Project: Challenges
+        CostCenter: NO PROGRAM / 000000  # resource-level cost center tags will override this default
```

Comment:
What would happen if, for some reason, a resource level tag was not applied or was missing?

Reply:
> What would happen if, for some reason, a resource level tag was not applied or was missing?

 Then we would categorize the resource as not being for a directly funded project.  At the end of the month we would let the account owner know how much activity in their account lacks a funding source.  If the amount is small then it's OK. The owner won't care.  If the amount is large, the account owner can question their team and perhaps add missing CostCenter tags.

Reply:
I'm not opposed your proposed approach but I'll share my concerns..

Ok, however I think you're making an assumption that we have some reporting mechanism in place to break down costs and intelligently give the account owner info about what's being tracked and what isn't.  The question is do we wan to accept `NO PROGRAM` until we have something like that in place?  

Also from my experience using AWS I will guarantee that there will be things that won't have tags on them even if you use cloudformation to provision resources.  Whether that's due to AWS bugs or the idiosyncratic ways that tags work in AWS.

Reply:
> I think you're making an assumption that we have some reporting mechanism in place to break down costs and intelligently give the account owner info about what's being tracked and what isn't.

Yes, none of this is useful until we have the script that that processes the AWS cost-and-usage report to provide rolled-up summaries to account owners and project PIs.  I have been tinkering with the AWS cost-and-usage report for a few months and am confident about our ability to "roll up" costs by account, cost center, user, etc. in a variety of ways.

> The question is do we wan[t] to accept NO PROGRAM until we have something like that in place?

I think so:  I'm not sure what else we can do besides (or in advance of) building and running the envisioned accounting script.  Is there an alternative we should consider?

> I will guarantee that there will be things that won't have tags on them even if you use cloudformation to provision resources.

Yes, I agree completely.  I imagine each time we roll up costs there will be a bunch of untagged 'dark matter' that we might need to dig into and analyze.  Until we see if the $ amount is small or large I don't know whether it will be a problem.

By the way, the reason I like the "NO PROGRAM" code at the resource level is that it allows us to differentiate between (1) a resource we explicitly say is not for a funded program and (2) an unexpected cost we aren't tracking.  It's a reasonable question to ask, "How much of (2) is there in our monthly bill?"


So we need to look at how much 'untagged' cost there is and (if it's significant) dive in to figure out why.  To me the strength of the "NO PROGRAM" tag is it lets us see the difference between items we explicitly say are for no particular program and items that we have just failed to tag.

Reply:
Apologies for a slight tangent.  I was going to try to test out some of the scripts found in this repo: https://github.com/ucsc-cgl/aws-reporting to see what the output was.  I'll report back with results.

Reply:
>  Is there an alternative we should consider

One alternative is to send the bill for the entire account to the responsible department.  We can either use `Department` tag, make up a department `CostCenter` tag, or even use the `AccountOwner` tag to identify who should get the bill.  Then let the individual department figure out specific resources costs themselves?   It looks  like @thomasyu888 is already looking for ways to slice and dice the costs.  There are also SAS services that can do this for us, no need to re-invent the wheel here.

Reply:
We will most likely host certain dev resources that will be utilized across by multiple projects so in those scenarios, it might even be difficult for us to split the costs, so we would probably just put it in "NO PROGRAM" code.  At the end of the day, maybe System Admins can try our best to make sure everything is tagged appropriately at the resource level. It'll be easier to tell once we get the costs of "untagged resources".

@zaro0508, agreed that we should re-use services when possible.  One thing to consider is that the system/account owners would potentially need access whatever billing information that allows us to use this service OR the reports are sent to us.  As an example, the script I linked above requires access to an S3 bucket (AWS billing can publish detailed billing results to an S3 bucket if we choose to).  Not saying we have to go that route, just some food for thought.